### PR TITLE
FIX: Can't change notification level of categories set to regular

### PR DESCRIPTION
**Pull request by tgxworld**
This commit fixes a bug on the client site where we would include the
`regular_category_ids` field when trying to update the notification levels of
categories for a user. The `regulary_category_ids` field should only be
included when the `mute_all_categories_by_default` is enabled

### CODE REVIEW

File: app/assets/javascripts/discourse/app/models/user.js [hunk 1]
```diff
@@ -399,12 +399,14 @@ const User = RestModel.extend({
 
     ["muted", "regular", "watched", "tracked", "watched_first_post"].forEach(
       (s) => {
```
Comment (martin-brennan) [inline app/assets/javascripts/discourse/app/models/user.js:401]: `s` is so undescriptive here... `trackingProp` or something would be better, especially since it's used multiple times. I don't even know what `s` could stand for (I know this is not you, but may as well improve at same time)

**Review by martin-brennan — APPROVED** [review]
Nice, only a minor comment :)

File: app/assets/javascripts/discourse/app/templates/preferences/categories.hbs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <div class="control-group category-notifications">
   <label class="control-label">{{i18n "user.categories_settings"}}</label>
 
-  <div class="controls tracking-controls">
+  <div class="controls tracking-controls tracking-controls--watched-categories">
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/templates/preferences/categories.hbs:4]: `--` is this related to any pattern we are following?

Comment (tgxworld) [reply app/assets/javascripts/discourse/app/templates/preferences/categories.hbs:4]: It represents the modifier syntax when following BEM

Comment (jjaffeux) [reply app/assets/javascripts/discourse/app/templates/preferences/categories.hbs:4]: I thought there was a space in this case (different class) but apparently not, ok then 🤷‍♂️ Swear

**Comment by tgxworld** [comment]
Thank you for reviewing @martin-brennan
